src/main/java/net/mcreator/ui/modgui/RangedItemGUI.java

```diff
@@ -133,16 +133,16 @@ public RangedItemGUI(MCreator mcreator, ModElement modElement, boolean editingMo
 
 		onBulletHitsBlock = new ProcedureSelector(this.withEntry("rangeditem/when_bullet_hits_block"), mcreator,
 				L10N.t("elementgui.ranged_item.event_bullet_hits_block"),
-				Dependency.fromString("x:number/y:number/z:number/world:world/entity:entity"));
+				Dependency.fromString("x:number/y:number/z:number/world:world/entity:entity/imediatesourceentity:entity"));
```

Comment:
I think it would be better to move these procedures to a new tab "Bullet procedures" so the UI is not too crammed

Reply:
I added a page for all the ranged item triggers, including the item ones